src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -169,22 +170,25 @@ public function set($id, $service)
         }
 
         if (isset($this->aliases[$id])) {
-            unset($this->aliases[$id]);
+            @trigger_error(sprintf('Replacing the "%s" (alias of "%s") service is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id, $this->aliases[$id]), E_USER_DEPRECATED);
+        } elseif (isset($this->services[$id])) {
```

Comment:
Not sure I'll answer well your question, but have a look to the [ProjectServiceContainer](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Tests/ContainerTest.php#L376-L431).

The `protected $services` is an empty array. `has` will however return true for the `->has('bar')` call, because the `getBarService()` method exists. `->get('bar')` will normally lazily create the service by calling the method, creating the instance and assign it to the service id. (However, it seems this container is missing the service id assignation in the `getBarService()`. Have a look to your `appDevDebugProjectContainer` eventually)

So, actually, a non-initialized service may not have its id in `$this->services`, thus `isset($this->services[$id])` won't be sufficient.